### PR TITLE
feat(shared): add complete model for NGSI-LD entity representation

### PR DESCRIPTION
**Pull request by bobeal**
- add complete models for attributes and attribute instances
- add JsonLdEntity to transport an expanded JSON-LD entity representation
- rename ExpandedEntity to NgsiLdEntity
- improve SoC between Json-LD / NGSI-LD / Json concerns
- rename NgsiLdParsingUtils to JsonLdUtils and only keep JSON-LD specific concerns in it
- move non related code to specific units (e.g. JsonUtils to parse JSON payloads)
- chore: misc tests cleanups

TODO:
- finish cleaning and refactoring unit tests in entity service

### CODE REVIEW

File: entity-service/src/main/kotlin/com/egm/stellio/entity/model/Attribute.kt [hunk 1]
```diff
@@ -47,28 +47,28 @@ open class Attribute(
     open fun serializeCoreProperties(): MutableMap<String, Any> {
         val resultEntity = mutableMapOf<String, Any>()
 
-        resultEntity[NgsiLdParsingUtils.NGSILD_CREATED_AT_PROPERTY] = mapOf(
-            NGSILD_ENTITY_TYPE to NGSILD_DATE_TIME_TYPE,
+        resultEntity[JsonLdUtils.NGSILD_CREATED_AT_PROPERTY] = mapOf(
```
Comment (vraybaud) [inline entity-service/src/main/kotlin/com/egm/stellio/entity/model/Attribute.kt:50]: Strange to have static imports and this not statically imported

File: entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt [hunk 2]
```diff
@@ -136,29 +117,27 @@ class EntityService(
...
-        propertyValues: Map<String, List<Any>>
+        ngsiLdPropertyInstance: NgsiLdPropertyInstance
     ): String {
-        val propertyValue = getPropertyValueFromMap(propertyValues, NGSILD_PROPERTY_VALUE)
-            ?: throw BadRequestDataException("Key $NGSILD_PROPERTY_VALUE not found in $propertyValues")
-
-        logger.debug("Creating property $propertyKey with value $propertyValue")
+        logger.debug("Creating property $propertyKey with value ${ngsiLdPropertyInstance.value}")
 
         val rawProperty = Property(
```
Comment (vraybaud) [inline entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:126]: ngsiLdPropertyInstance.toProperty() could be really classy, no?

Comment (vraybaud) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:126]: toRawProperty() ?

File: entity-service/src/main/kotlin/com/egm/stellio/entity/model/Attribute.kt (hunk 1 above)
Comment (bobeal) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/model/Attribute.kt:50]: Well, just some legacy (nothing forces you to static import and we have no defined rule about this)

File: entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt [hunk 3]
```diff
@@ -426,89 +383,99 @@ class EntityService(
...
-                    logger.info("GeoProperty ${it.key} already exists on $entityId and overwrite is not allowed, ignoring")
-                    listOf(Triple(
-                        it.key,
-                        false,
-                        "GeoProperty ${it.key} already exists on $entityId and overwrite is not allowed, ignoring"
-                    ))
                 } else {
-                    updateLocationPropertyOfEntity(entityRepository.findById(entityId).get(), it.key, attributeValue[0])
-                    listOf(Triple(it.key, true, null))
+                    // TODO we should avoid this fake else
```
Comment (vraybaud) [inline entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:474]: you don't want to remove it in this PR ?

File: entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt [hunk 4]
```diff
@@ -528,188 +495,61 @@ class EntityService(
...
+                    NotUpdatedDetails(shortAttributeName, e.message)
                 )
             }
         }
 
-        if (updatedAttributesPayload.isNotEmpty()) {
-            val entityType = entity.type[0].extractShortTypeFromExpanded()
-            updatedAttributesPayload.forEach {
-                val entityEvent = EntityEvent(EventType.UPDATE, entity.id, entityType, it, null)
-                applicationEventPublisher.publishEvent(entityEvent)
```
[the remark this answers is not in the document]
Comment (bobeal) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:587]: Discussed in a daily meeting.

**Pull request merged**